.gitlab-ci.yml

```diff
@@ -206,45 +204,318 @@ variables:
+    when: never
+  - when: manual
+    allow_failure: true
 
-.if_not_version_6: &if_not_version_6
-  if: $RELEASE_VERSION_6 == ""
+.on_dev_branch_a7_manual:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_master_branch
+    when: never
+  - <<: *if_release_branch
+    when: never
+  - <<: *if_tagged_commit
+    when: never
+  - when: manual
+    allow_failure: true
 
-.if_version_7: &if_version_7
-  if: $RELEASE_VERSION_7 != ""
+.on_dev_branch_all_builds_a6_manual:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_master_branch
+    when: never
+  - <<: *if_release_branch
+    when: never
+  - <<: *if_tagged_commit
+    when: never
+  - <<: *if_run_all_builds
+    when: manual
+    allow_failure: true
 
-.if_not_version_7: &if_not_version_7
-  if: $RELEASE_VERSION_7 == ""
+.on_dev_branch_all_builds_a7_manual:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_master_branch
+    when: never
+  - <<: *if_release_branch
+    when: never
+  - <<: *if_tagged_commit
+    when: never
+  - <<: *if_run_all_builds
+    when: manual
+    allow_failure: true
+
+.on_master:
+  - <<: *if_master_branch
+
+.on_master_manual:
+  - <<: *if_master_branch
+    when: manual
+    allow_failure: true
+
+.on_master_a6:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_master_branch
+
+.on_master_a7:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_master_branch
+
+.on_tag_or_a7:
+  - <<: *if_tagged_commit
+  - <<: *if_version_7
+
+.on_tag_or_a7_all_builds:
+  - <<: *if_not_run_all_builds
+    when: never
+  - <<: *if_tagged_commit
+  - <<: *if_version_7
+
+.on_deploy:
+  - <<: *if_deploy
+
+.on_deploy_failure:
+  - <<: *if_deploy
+    when: on_failure
+
+.on_deploy_a6:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_deploy
+
+.on_deploy_a6_manual:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_deploy
+    when: manual
+    allow_failure: true
+
+.on_deploy_a7:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_deploy
+
+.on_deploy_a7_manual:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_deploy
+    when: manual
+    allow_failure: true
+
+.on_deploy_nightly_repo_branch_a6:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_not_nightly_repo_branch
+    when: never
+  - <<: *if_deploy
+
+.on_deploy_nightly_repo_branch_a7:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_not_nightly_repo_branch
+    when: never
+  - <<: *if_deploy
+
+.on_deploy_stable_or_beta_repo_branch:
+  - <<: *if_not_stable_or_beta_repo_branch
+    when: never
+  - <<: *if_deploy
+
+.on_deploy_stable_or_beta_repo_branch_a6:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_not_stable_or_beta_repo_branch
+    when: never
+  - <<: *if_deploy
+
+.on_deploy_stable_or_beta_repo_branch_a6_manual:
+  - <<: *if_not_version_6
+    when: never
+  - <<: *if_not_stable_or_beta_repo_branch
+    when: never
+  - <<: *if_deploy
+    when: manual
+    allow_failure: true
+
+.on_deploy_stable_or_beta_repo_branch_a6_always:
```

Comment:
What's the difference in behavior between this and ` .on_deploy_stable_or_beta_repo_branch_a6`?

Reply:
The difference between `.<rule>` and `.<rule>_manual`, `.<rule>_always` is the `when` condition: on `.<rule>`, the default (`when: on_success`) is used; on `.<rule>_manual`, `when: manual` is specified; and on `.<rule>_always`, `when: always` is specified. A job with `when: always` will always run, even if previous jobs in its dependency tree failed.

Reply:
Gotcha, thanks